### PR TITLE
sci-libs/hipFFT: ROCm FFT marshalling library

### PR DESCRIPTION
**Pull request by littlewu2508**
hipFFT is the front end of rocFFT, and is dependency of ROCm
supported math/DL frameworks like pytorch.

Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

**Comment by heroxbd** [comment]
This is new package, have you asked @candrews and @justxi for permissions to put their names into the `metadata.xml`.  Do you want to be the maintainer of this new package?

**Comment by justxi** [comment]
You can add my name to the metadata.xml.

**Comment by littlewu2508** [comment]
> 
> 
> This is a new package, have you asked @candrews and @justxi for permissions to put their names into the `metadata.xml`. Do you want to be the maintainer of this new package?

I should've ask their permissions, especially for hipFFT which seems not in https://github.com/justxi/rocm, So I should put myself into the maintainer list.

File: sci-libs/hipFFT/hipFFT-4.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr"
+		-DCMAKE_INSTALL_INCLUDEDIR="include/hipfft"
+		-DCMAKE_SKIP_RPATH=ON
+		-DBUILD_CLIENTS_TESTS=$(usex test ON OFF)
+		-DBUILD_CLIENTS_RIDER=$(usex benchmark ON OFF)
+		-D__skip_rocmclang="ON" ## fix cmake-3.21 configuration issue caused by officialy support programming language "HIP"
+	)
+	[ -n "${AMDGPU_TARGETS}" ] && mycmakeargs+=( -DAMDGPU_TARGETS="${AMDGPU_TARGETS}" )
```
Comment (heroxbd) [inline sci-libs/hipFFT/hipFFT-4.3.0.ebuild:52]: New trick please.

File: sci-libs/hipFFT/hipFFT-4.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+S="${WORKDIR}/hipFFT-rocm-${PV}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-4.3.0-gentoo-install-locations.patch"
+	"${FILESDIR}/${PN}-4.3.0-remove-git-dependency.patch"
+	"${FILESDIR}/${PN}-4.3.0-add-complex-header.patch"
+)
+
+src_prepare() {
+	use test && rmdir rocFFT && ln -s ../rocFFT-rocm-${PV} rocFFT
```
Comment (heroxbd) [inline sci-libs/hipFFT/hipFFT-4.3.0.ebuild:34]: Should call `|| die` here.

```
if use test; then
    rmdir rocFFT || die
    ln -s ../rocFFT-rocm-${PV} rocFFT || die
fi

**Review by heroxbd — CHANGES_REQUESTED** [review]
From README.md, 

### Client build dependencies

* The clients (samples, tests etc) included with the hipFFT source
  depend on FFTW, gtest, and boost program-options.

So fftw and boost need to be in BDEPEND when USE=test.

**Comment by heroxbd** [comment]
@littlewu2508 let's remove @candrews 's name from `metadata.xml` for the moment to get this package included.

@candrews feel free to add yourself to the maintainers list in the future.  Thanks!

**Comment by littlewu2508** [comment]
> @littlewu2508 let's remove @candrews 's name from `metadata.xml` for the moment to get this package included.
 
Done. I also added my name in the maintainer list.

**Pull request closed without merging**